eth/stagedsync/stage_execute_zkevm.go

```diff
@@ -125,8 +125,18 @@ func SpawnExecuteBlocksStageZk(s *StageState, u Unwinder, tx kv.RwTx, toBlock ui
 
 	initialBlock := stageProgress + 1
 	eridb := erigon_db.NewErigonDb(tx)
+
+	prevheaderHash, err := rawdb.ReadCanonicalHash(tx, stageProgress)
+	if err != nil {
+		return err
+	}
+	header, err := cfg.blockReader.Header(ctx, tx, prevheaderHash, stageProgress)
+	if err != nil {
+		return err
+	}
+	prevBlockHash := header.Root
 Loop:
-	for blockNum := stageProgress + 1; blockNum <= to; blockNum++ {
+	for blockNum := stageProgress + 1; blockNum <= 59057; blockNum++ {
```

Comment:
do we want to keep this limitation in the loop?

Reply:
nope :D